tests/e2e/test_async_client.py

```diff
@@ -36,6 +36,7 @@ def setUp(self) -> None:
     async def test_get_status(self):
         query = Query(name="No Name", query_id=1276442, params=[])
         dune = AsyncDuneClient(self.valid_api_key)
+        await dune.connect()
```

Comment:
I would really prefer if these kinds of things can be put directly into the constructor. Users shouldn't have to deal with all of this manually. This an close session. Why do we have to do all this stuff?